tests/unit/cpp/test_inet.cpp

```diff
@@ -318,273 +318,312 @@ TEST(etcpal_cpp_inet, ip_from_string_works)
+
+  // Construct from raw IPv6 data, a scope ID and port number.
+  const etcpal::SockAddr sockaddr3(s_v6_data.data(), 12, 0xdead);
+
+  TEST_ASSERT_TRUE(sockaddr3.IsValid());
+  TEST_ASSERT_TRUE(sockaddr3.IsV6());
+  TEST_ASSERT_EQUAL_UINT8_ARRAY(sockaddr3.v6_data(), s_v6_data.data(), ETCPAL_IPV6_BYTES);
+  TEST_ASSERT_EQUAL(sockaddr3.scope_id(), 12);
+  TEST_ASSERT_EQUAL_UINT16(sockaddr3.port(), 0xdead);
+
+  // Construct from an IpAddr and port number
+  const etcpal::SockAddr sockaddr4(etcpal::IpAddr(0xdeadbeef), 0xfeed);
+  TEST_ASSERT_TRUE(sockaddr4.IsValid());
+  TEST_ASSERT_TRUE(sockaddr4.IsV4());
+  TEST_ASSERT_EQUAL_UINT32(sockaddr4.v4_data(), 0xdeadbeef);
+  TEST_ASSERT_EQUAL_UINT16(sockaddr4.port(), 0xfeed);
 }
 
 TEST(etcpal_cpp_inet, sockaddr_to_string_works)
 {
-    const etcpal::SockAddr sockaddr_v4(etcpal::IpAddr::FromString("10.101.2.3"), 5555);
-    TEST_ASSERT_EQUAL_STRING(sockaddr_v4.ToString().c_str(), "10.101.2.3:5555");
+  const etcpal::SockAddr sockaddr_v4(etcpal::IpAddr::FromString("10.101.2.3"), 5555);
+  TEST_ASSERT_EQUAL_STRING(sockaddr_v4.ToString().c_str(), "10.101.2.3:5555");
+
+  const etcpal::SockAddr sockaddr_v6(etcpal::IpAddr::FromString("2001:db8::2222:3333"), 6666);
+  auto sockaddr_str = sockaddr_v6.ToString();
+  ConvertStringToLowercase(sockaddr_str);
+  TEST_ASSERT_EQUAL_STRING(sockaddr_str.c_str(), "[2001:db8::2222:3333]:6666");
+
+  const etcpal::SockAddr sockaddr_invalid;
+  TEST_ASSERT_TRUE(sockaddr_invalid.ToString().empty());
+}
+
+TEST(etcpal_cpp_inet, sockaddr_is_link_local_works)
+{
+  const auto sockaddr = etcpal::SockAddr(etcpal::IpAddr::FromString("169.254.20.40"), 8888);
+  TEST_ASSERT_TRUE(sockaddr.IsLinkLocal());
+}
+
+TEST(etcpal_cpp_inet, sockaddr_is_loopback_works)
+{
+  const auto sockaddr = etcpal::SockAddr(etcpal::IpAddr::FromString("::1"), 4000);
+  TEST_ASSERT_TRUE(sockaddr.IsLoopback());
+}
+
+TEST(etcpal_cpp_inet, sockaddr_is_multicast_works)
+{
+  const auto sockaddr = etcpal::SockAddr(etcpal::IpAddr::FromString("224.127.0.1"), 1234);
+  TEST_ASSERT_TRUE(sockaddr.IsMulticast());
+}
 
-    const etcpal::SockAddr sockaddr_v6(etcpal::IpAddr::FromString("2001:db8::2222:3333"), 6666);
-    auto sockaddr_str = sockaddr_v6.ToString();
-    ConvertStringToLowercase(sockaddr_str);
-    TEST_ASSERT_EQUAL_STRING(sockaddr_str.c_str(), "[2001:db8::2222:3333]:6666");
+TEST(etcpal_cpp_inet, sockaddr_is_wildcard_works)
+{
+  auto wildcard = etcpal::SockAddr(etcpal::IpAddr::WildcardV4(), 2000);
+  TEST_ASSERT_TRUE(wildcard.IsWildcard());
+  wildcard = etcpal::SockAddr(etcpal::IpAddr::WildcardV6(), 3000);
+  TEST_ASSERT_TRUE(wildcard.IsWildcard());
+}
 
-    const etcpal::SockAddr sockaddr_invalid;
-    TEST_ASSERT_TRUE(sockaddr_invalid.ToString().empty());
+TEST(etcpal_cpp_inet, sockaddr_to_ip_works)
+{
+  const auto sockaddr = etcpal::SockAddr(0x0a0b0c0d, 8888);
+  const auto ip = sockaddr.ip();
+  TEST_ASSERT_TRUE(ip.IsV4());
+  TEST_ASSERT_EQUAL_UINT32(ip.v4_data(), 0x0a0b0c0d);
```

Comment:
This is a super minor thing, but I would put 0x0a0b0c0d into a local const int and use that in lines 509 and 512.